clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -299,34 +289,52 @@ private TaskHealth inferHealthNetworkBindedContainer(
     return overallTaskHealth;
   }
 
+  private TargetHealthDescription findHealthDescription(
+      List<TargetHealthDescription> targetHealths, String targetId, Integer targetPort) {
+
+    return targetHealths.stream()
+        .filter(
+            h ->
+                h.getTarget().getId().equals(targetId)
+                    && h.getTarget().getPort().equals(targetPort))
+        .findFirst()
+        .orElse(null);
+  }
+
   private TaskHealth describeTargetHealth(
-      AmazonElasticLoadBalancing amazonloadBalancing,
+      TargetHealthCacheClient targetHealthCacheClient,
       Task task,
       Service loadBalancerService,
       String serviceName,
       String targetGroupArn,
       String targetId,
       Integer targetPort,
       TaskHealth overallTaskHealth) {
-    DescribeTargetHealthResult describeTargetHealthResult =
-        amazonloadBalancing.describeTargetHealth(
-            new DescribeTargetHealthRequest()
-                .withTargetGroupArn(targetGroupArn)
-                .withTargets(new TargetDescription().withId(targetId).withPort(targetPort)));
-
-    if (describeTargetHealthResult.getTargetHealthDescriptions().isEmpty()) {
-      log.debug("Target health description is empty");
+
+    String targetHealthKey = Keys.getTargetHealthKey(accountName, region, targetGroupArn);
+    EcsTargetHealth targetHealth = targetHealthCacheClient.get(targetHealthKey);
+
+    if (targetHealth == null) {
+      log.debug("Cached EcsTargetHealth is empty for targetGroup {}", targetGroupArn);
+      evictStaleData(task, loadBalancerService);
+      return overallTaskHealth;
+    }
+    TargetHealthDescription targetHealthDescription =
+        findHealthDescription(targetHealth.getTargetHealthDescriptions(), targetId, targetPort);
+
+    if (targetHealthDescription == null) {
+      log.debug(
+          "TargetHealthDescription is empty on targetGroup '{}' for {}:{}",
+          targetGroupArn,
+          targetId,
+          targetPort);
       evictStaleData(task, loadBalancerService);
```

Comment:
tbh I'm not sure why we do these evictions of the service and task definition here, but should we be evicting the target group health too?  Really not sure what impact that would have

Reply:
yeah, I'm not sure what role evictions have if the cache is being rebuilt (as in, described with the SDK and fully re-populated) every time the agent executes - but maybe that's not truly what's going on? I'll need to poke around some.

Reply:
The eviction would only make sense if you're calling AWS, it would remove the stale entries from the cache. Don't think you need it anymore.

Reply:
per offline discussion: punting on reevaluating evictions for a future PR